test/Constraints/overload.swift

```diff
@@ -287,7 +287,22 @@ func rdar97396399() {
 }
 
 // https://github.com/apple/swift/issues/63834
-func f63834(int: Int, string: String) {} // expected-note{{found candidate with type '(Int, String) -> ()'}}
-func f63834(int: Int, string: Bool) {} // expected-note{{found candidate with type '(Int, Bool) -> ()'}}
+func f63834(int: Int, string: String) {} // expected-note 3{{found candidate with type '(Int, String) -> ()'}}
+func f63834(int: Int, string: Bool) {} // expected-note 3{{found candidate with type '(Int, Bool) -> ()'}}
 
-let a = f63834(int:string:) as (Int, Int) -> Void // expected-error{{no exact matches in reference to global function 'f63834'}}
+func f63834_1(int: Int, string: Bool) {} // expected-note{{candidate '(Int) -> Void' requires 1 argument, but 2 were provided}}
```

Comment:
Sorry to only notice this now when you pushed and re-triggered the tests. I think this warrants a FIXME. This is better than a generic error, but the message is still far from optimal. We are diagnosing the function reference instead of the coercion type. It ought to be the other way around, e.g. `coercion type '(Int) -> Void' requires 1 argument, but candidate requires 2`.